nexus/src/db/datastore/vpc.rs

```diff
@@ -128,33 +130,80 @@ impl DataStore {
     pub async fn project_delete_vpc(
         &self,
         opctx: &OpContext,
+        db_vpc: &Vpc,
         authz_vpc: &authz::Vpc,
     ) -> DeleteResult {
         opctx.authorize(authz::Action::Delete, authz_vpc).await?;
 
         use db::schema::vpc::dsl;
+        use db::schema::vpc_subnet;
 
         // Note that we don't ensure the firewall rules are empty here, because
         // we allow deleting VPCs with firewall rules present. Inserting new
         // rules is serialized with respect to the deletion by the row lock
         // associated with the VPC row, since we use the collection insert CTE
         // pattern to add firewall rules.
 
+        // We _do_ need to check for the existence of subnets. VPC Subnets
+        // cannot be deleted while there are network interfaces in them
+        // (associations between an instance and a VPC Subnet). Because VPC
+        // Subnets are themselves containers for resources that we don't want to
+        // auto-delete (now, anyway), we've got to check there aren't any. We
+        // _might_ be able to make this a check for NICs, rather than subnets,
+        // but we can't have NICs be a child of both tables at this point, and
+        // we need to prevent VPC Subnets from being deleted while they have
+        // NICs in them as well.
+        println!("ABOUT TO SEARCH FOR SUBNETS");
```

Comment:
Probably should remove the `println`s, here and elsewhere in the PR, before merging

Reply:
We should indeed! Done in d2b3ce84.